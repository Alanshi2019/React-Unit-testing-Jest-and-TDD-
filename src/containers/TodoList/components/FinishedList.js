import React, { Component } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheckCircle } from "@fortawesome/free-solid-svg-icons";

class FinishedList extends Component {
  render() {
    const { List, deleteItem } = this.props;
    return (
      <div className="undo-list">
        <div className="undo-list-title">
          Finished
          <div data-test="count" className="undo-list-count">
            {List.length}
          </div>
        </div>

        <ul className="undo-list-ul">
          {List.map((item, index) => {
            return (
              <li
                className="finished-list-li"
                data-test="list-item"
                key={index}
              >
                <span data-test="check-item" className="checkbox">
                  <FontAwesomeIcon className="icon" icon={faCheckCircle} />
                </span>
                {item}
                <span
                  data-test="delete-item"
                  className="finished-item-remove"
                  onClick={e => {
                    e.stopPropagation();
                    deleteItem(index);
                  }}
                >
                  -
                </span>
              </li>
            );
          })}
        </ul>
      </div>
    );
  }
}

export default FinishedList;
