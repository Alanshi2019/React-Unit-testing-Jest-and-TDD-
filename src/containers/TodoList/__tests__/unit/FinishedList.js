import React from "react";
import { shallow } from "enzyme";
import FinishedList from "../../components/FinishedList";
import { findTestWrapper } from "../../../../utils/testUtils";

it("FinishedList render", () => {
  const wrapper = shallow(<FinishedList List={[]} />);
  expect(wrapper).toMatchSnapshot();
}); // 这个测试的目的在于 相当于是锁死了UI 只要发生一点变化 就会报错

it("FinishedList init count = 0, List is enmpyt ", () => {
  const wrapper = shallow(<FinishedList List={[]} />);
  const countElem = findTestWrapper(wrapper, "count");
  const listElem = findTestWrapper(wrapper, "list-item");
  expect(countElem.text()).toEqual("0");
  expect(listElem.length).toEqual(0);
});
it("FinishedList is not empyt, count changing, List changing", () => {
  const wrapper = shallow(
    <FinishedList
      List={["First thing I need to do", "Second thing I need to do"]}
    />
  );
  const countElem = findTestWrapper(wrapper, "count");
  const listElem = findTestWrapper(wrapper, "list-item");
  expect(countElem.text()).toEqual("2");
  expect(listElem.length).toEqual(2);
});

it("FinishedList is not empty, RemoveButtons exist.", () => {
  const wrapper = shallow(
    <FinishedList
      List={["First thing I need to do", "Second thing I need to do"]}
    />
  );
  const deleteElem = findTestWrapper(wrapper, "delete-item");
  expect(deleteElem.length).toEqual(2);
});

it("FinishedList is not empty RemoveButtons onClick check.", () => {
  const fn = jest.fn();
  const index = 0;
  const wrapper = shallow(
    <FinishedList
      List={["First thing I need to do", "Second thing I need to do"]}
      deleteItem={fn}
    />
  );
  const deleteElem = findTestWrapper(wrapper, "delete-item");
  deleteElem.at(index).simulate("click", { stopPropagation: () => {} });
  expect(fn).toHaveBeenLastCalledWith(index);
});
