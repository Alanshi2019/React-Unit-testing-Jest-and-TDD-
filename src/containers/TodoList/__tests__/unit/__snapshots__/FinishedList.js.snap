// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinishedList render 1`] = `
<div
  className="undo-list"
>
  <div
    className="undo-list-title"
  >
    Finished
    <div
      className="undo-list-count"
      data-test="count"
    >
      0
    </div>
  </div>
  <ul
    className="undo-list-ul"
  />
</div>
`;
